### PR TITLE
speedup sphinx build with caching pip

### PR DESCRIPTION
**Pull request by smoench**

### CODE REVIEW

**Comment by ricknox** [comment]
cc: @OskarStark

**Comment by javiereguiluz** [comment]
@smoench this is really nice. Thanks for all your help to improve the performance of the doc linter.

**Review by wouterj — APPROVED** [review]

**Comment by wouterj** [comment]
Thanks a lot!

In the CMF docs, we also cached `$HOME/.cache/pip`, to speed up pip. Maybe that's an option here as well?

**Comment by smoench** [comment]
Sure, I was also thinking about that

**Review by OskarStark — APPROVED** [review]
Nice Improvement ✌🏻

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -29,12 +29,26 @@ jobs:
...
+            -   name: "Cache pip"
+                uses: actions/cache@v1
+                with:
+                    path: ~/.cache/pip
+                    key: ${{ runner.os }}-pip-${{ hashFiles('_build/.requirements.txt') }}
+                    restore-keys: |
+                        ${{ runner.os }}-pip-
+
             -   name: "Install Sphinx"
                 run: pip install --user sphinx
```
Comment (smoench) [inline .github/workflows/ci.yaml]: @OskarStark Why is this needed? Because `_build/.requirements.txt` has also a reference for sphinx but in version 1.3.6

**Comment by smoench** [comment]
I don't know what causes following error:
```
Warning, treated as error:
/home/runner/work/symfony-docs/symfony-docs/form/form_customization.rst:None: WARNING: duplicate label form-customization-sidebar, other instance in /home/runner/work/symfony-docs/symfony-docs/form/form_themes.rst
```

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (OskarStark) [reply .github/workflows/ci.yaml]: Not sure, let’s try without...
Initially I used the same approach as on Travis

Comment (wouterj) [reply .github/workflows/ci.yaml]: It wasn't part of the travis config: https://github.com/symfony/symfony-docs/pull/13061/files#diff-354f30a63fb0907d4ad57269548329e3

So I guess it has something to do with GitHub actions or otherwise, can be removed

Comment (OskarStark) [reply .github/workflows/ci.yaml]: Hmm strange, maybe it was part of .symfony.cloud.yaml ?

Comment (smoench) [reply .github/workflows/ci.yaml]: I removed it, works fine.

**Comment by wouterj** [comment]
> I don't know what causes following error:

Hmm, that reference target (`form-customization-sidebar`) lives in `form/form_customization.rst` in 3.4 and in `form/form_themes.rst` for the newer versions. Seems like the cache is maybe shared among all branches, resulting in this error? Or does the docs somehow render a wrong version of the documentation?

**Comment by smoench** [comment]
On the first try the base branch was on `master` but changed I changed it to `3.4`. I thought it wouldn't matter as of the cache restrictions https://help.github.com/en/actions/configuring-and-managing-workflows/caching-dependencies-to-speed-up-workflows#restrictions-for-accessing-a-cache.
I changed cache keys now.

But some how after the cache hit and cache file has been extracted, sphinx thinks all files has been changed. So cache does not work properly 🤷‍♂

**Comment by OskarStark** [comment]
So we only go with the pip cache and not the other ones?

**Comment by smoench** [comment]
Unfortunately, yes! But perhaps someone can confirm that it is working with a more latest sphinx version than 1.3.6.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -29,12 +29,29 @@ jobs:
...
+            -   name: "Cache pip"
+                uses: actions/cache@v1
+                with:
+                    path: ~/.cache/pip
+                    key: ${{ runner.os }}-pip-${{ hashFiles('_build/.requirements.txt') }}
+                    restore-keys: |
+                        ${{ runner.os }}-pip-
 
-            -   name: "Install custom requirements via pip"
+            -   name: "Install spinx + requirements via pip"
```
[the remark this answers is not in the document]
Comment (smoench) [reply .github/workflows/ci.yaml]: done

**Comment by smoench** [comment]
Well, Sphinx uses `mtime` (in all versions) for detecting changes. This works on a local machine but not in a CI that makes a fresh checkout.

**Review by OskarStark — APPROVED** [review]

**Comment by OskarStark** [comment]
Thank you Simon.
